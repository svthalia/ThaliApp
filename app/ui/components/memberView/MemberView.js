import React from 'react';
import PropTypes from 'prop-types';
import {
  ImageBackground,
  StyleSheet,
  Text,
  TouchableHighlight,
  ViewPropTypes,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import styles from './style/MemberView';
import SquareView from './SquareView';

const MemberView = (props) => (
  <SquareView style={props.style} size={props.size}>
    <TouchableHighlight
      style={styles.image}
      onPress={() => props.loadProfile(props.member.pk)}
    >
      <ImageBackground style={styles.image} source={{ uri: props.member.photo }}>
        <LinearGradient colors={['#55000000', '#000000']} style={styles.overlayGradient} />
        <Text style={styles.nameText}>
          {props.member.name}
        </Text>
      </ImageBackground>
    </TouchableHighlight>
  </SquareView>
);

MemberView.propTypes = {
  member: PropTypes.shape({
    pk: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    photo: PropTypes.string.isRequired,
  }).isRequired,
  size: PropTypes.number.isRequired,
  style: ViewPropTypes.style,
  loadProfile: PropTypes.func.isRequired,
};

const defaultStyles = StyleSheet.create({
});

MemberView.defaultProps = {
  style: defaultStyles,
};

export default MemberView;
