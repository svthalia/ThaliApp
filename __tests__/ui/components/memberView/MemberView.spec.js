import React from 'react';
import renderer from 'react-test-renderer';
import MemberView from '../../../../app/ui/components/memberView/MemberView';

jest.mock('react-navigation', () => ({
  withNavigation: component => component,
}));

jest.mock('react-native-snackbar', () => ({
  show: jest.fn(),
}));

describe('MemberView component', () => {
  const initialState = {
    session: {
      token: 'token123',
    },
  };

  const member = {
    pk: 1,
    name: 'Lorem ipsum',
    photo: 'http://example.org/example.png',
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemberView store={initialState} member={member} size={20} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
