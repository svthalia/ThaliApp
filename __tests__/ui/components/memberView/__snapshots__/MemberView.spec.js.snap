// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberView component renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "height": 20,
        "width": 20,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://example.org/example.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <BVLinearGradient
        colors={
          Array [
            5570560,
            4278190080,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0.6,
            "position": "absolute",
            "right": 0,
            "top": "50%",
          }
        }
      />
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "bottom": 4,
            "color": "#FFFFFF",
            "fontSize": 11,
            "left": 6,
            "position": "absolute",
            "right": 6,
          }
        }
      >
        Lorem ipsum
      </Text>
    </View>
  </View>
</View>
`;
